common/shareddb.cpp

```diff
@@ -774,27 +769,26 @@ bool SharedDatabase::GetInventory(uint32 account_id, char *name, EQ::InventoryPr
 		return false;
 	}
 
-	for (auto row = results.begin(); row != results.end(); ++row) {
+	for (auto& row = results.begin(); row != results.end(); ++row) {
 		int16 slot_id = atoi(row[0]);
 		uint32 item_id = atoi(row[1]);
-		int8 charges = atoi(row[2]);
-		uint32 color = atoul(row[3]);
+		const int8 charges = atoi(row[2]);
+		const uint32 color = atoul(row[3]);
 
 		uint32 aug[EQ::invaug::SOCKET_COUNT];
-		aug[0] = (uint32)atoi(row[4]);
-		aug[1] = (uint32)atoi(row[5]);
-		aug[2] = (uint32)atoi(row[6]);
-		aug[3] = (uint32)atoi(row[7]);
-		aug[4] = (uint32)atoi(row[8]);
-		aug[5] = (uint32)atoi(row[9]);
-
-		bool instnodrop = (row[10] && (uint16)atoi(row[10])) ? true : false;
-		uint32 ornament_icon = std::stoul(row[12]);
-		uint32 ornament_idfile = std::stoul(row[13]);
+		aug[0] = static_cast<uint32>(atoi(row[4]));
+		aug[1] = static_cast<uint32>(atoi(row[5]));
+		aug[2] = static_cast<uint32>(atoi(row[6]));
+		aug[3] = static_cast<uint32>(atoi(row[7]));
+		aug[4] = static_cast<uint32>(atoi(row[8]));
+		aug[5] = static_cast<uint32>(atoi(row[9]));
```

Comment:
With casts like this, should we just use std::stoul? Versus casting to int then back to unsigned.

Reply:
I had thought about that, but functionality would technically change since stoul will error if input is invalid where atoi will not and will return 0 instead.

Reply:
Could use `atoul` at that point if you want the default 0. Just doesn't make sense to pull as signed then cast to unsigned.

Reply:
atoul sounds like a good solution to me, doesn't change functionality but fixes the code smells. I'll make the change when I have time at home.

Reply:
Sounds good. There were more places doing this than just right here I believe.